src/mlpack/methods/ann/layer/softshrink_impl.hpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file softshrink.hpp
+ * @author Lakshya Ojha
+ *
+ * Implementation of Soft Shrink activation function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_SOFTSHRINK_IMPL_HPP
+
+// In case it hasn't yet been included
+#include "softshrink.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+// This constructor is called for Soft Shrink activation function.
+// lambda is a hyperparameter.
+template<typename InputDataType, typename OutputDataType>
+SoftShrink<InputDataType, OutputDataType>::SoftShrink(const double lambda) :
+    lambda(lambda)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename OutputType>
+void SoftShrink<InputDataType, OutputDataType>::Forward(
+    const InputType&& input, OutputType&& output)
+{
+  Fn(input, output);
+}
```

Comment:
Looks like you have implemented the Soft Shrink Function in softshrink.hpp itself.
Ideally the implementation should be inside softshink_impl.hpp

So you could move your code from Fn() to here 
```
void SoftShrink<InputDataType, OutputDataType>::Forward(
    const InputType&& input, OutputType&& output)
{
//here
}
```
and delete the Fn() itself.

Refer some other layers `layername_impl.hpp` and `layername.hpp` from `src/mlpack/methods/ann/layer` to see how they are implemented.
For example: [c_relu_impl.hpp](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/c_relu_impl.hpp) check out the Forward() and Backward() functions

Reply:
Hi @Hemal-Mamtora, I am not sure if that is necessary. However if you feel it is there are some PR open with similar issues.
I think the reason why this was done across multiple open PRs is: 
1. Makes code similar to activation functions in activation_functions directory. 
2. Also for future contributions of activation functions contributors can easily gain insight of what needs to be implemented.
3. Similarity to other activation function layers such as 
       1. elu 
       2. leaky_relu 
       3. parametric_relu
However, I think changes that you suggested can be made easily.

Reply:
@kartikdutt18, thanks for pointing out!
I see, all the files under `/activation_functions/` are formatted as done in this PR. 

Also that there is no disctinction like `sample_activation.hpp` and `sample_activation_impl.hpp`

But when I see `/layers/` folder and the `/loss_functions/` folder, all are in the format that I described in the comments above.

Now I am not sure, which format to follow.
@zoq , perhaps you could provide some clarity.

Reply:
I agree with @kartikdutt18  on this one. 
I believe that following the implementation styles that have already been used in previously accepted functions like above mentioned is a better way to do here also.
I have taken great care to be with the flow of all other functions because it makes it easy to understand any new functions that is to be implemented.
Also do tell me about what you think about the same.
Thanks.

Reply:
Either is fine for me, I guess if we would put it into the `Forward` function, we could save some more space, and someone who likes to look into the implementation could directly see it. Let me know what you think. I guess, if we put it into the `Forward` function, we should probably adjust the other layer as well, to be more consistent. But that should be straightforward.